### ISSUE TITLE
Isolation scope name problems

### ISSUE THREAD

**Issue by whitecolor** [staltz/cycle-onionify#8]
### Problem 1
How to be when scope named as state property is not enough:

```js
// this needs 'child' from onion state
isolate(Child, 'child')({onion, HTTP)

// and this needs 'child' from onion state, 
// but HTTP source needs different scope not to mess with `Child`
isolate(AnotherChild, 'child')({onion, HTTP) 
```

### Problem 2
What if one need to pass more nested property to child like "some.nested.child"

```js
// with mapping I would pass it like:
onion.state$.map(state => state.some.nested.child)
```

**Comment by staltz** [comment]
I think problem 1 can be solved with something like this: https://github.com/staltz/cycle-onionify/issues/3
It also depends if Child and AnotherChild just *read* state but don't write it. If that's the case, then `state$` can be passed instead of `sources.onion`.

Problem 2:
It depends, if it's read-only state, we can do like I suggested above. Otherwise, we could consider supporting an API like `isolate(Child, 'some.nested.child')`

**Comment by whitecolor** [comment]
`'some.nested.child'` it is still imperfect solution, as object property itself may contain dot.

I like that onionify tries to unify ideas for managing state, though currently I prefer more transparent and custom solution, something that I described in recent gist: https://gist.github.com/whitecolor/0f015856a335ca418e22d4bba4d97d68

**Comment by staltz** [comment]
> as object property itself may contain dot

I've never seen a single case of this. :)

**Comment by davidskuza** [comment]
What about `obj['some.dots']`?

**Comment by jakelazaroff** [comment]
I've also seen arrays of strings/numbers used to denote a nested path, as in `isolate(Child, ['some', 'nested', 'child'])`. It's a little noisier than a dot-delimited string but it works if you're really worried about having dots in property names.

**Comment by staltz** [comment]
The lenses proposal in #11 would solve this issue entirely.

**Comment by staltz** [comment]
Solved by issue #11
